src/care/LoopFuser.h

```diff
@@ -339,6 +339,266 @@ class SerializableDeviceLambda {
+///////////////////////////////////////////////////////////////////////////
+class FusedActionsObserver : public FusedActions {
+public:
+   CARE_DLL_API static FusedActionsObserver * activeObserver;
+
+   FusedActionsObserver() : FusedActions(),
+                            m_fused_actions(),
+                            m_fused_action_order(),
+                            m_last_insert_priority(-FLT_MAX),
+                            m_to_be_freed(),
+                            m_recording(false) {
+    }
+
+   void startRecording() {
+      for (auto & action_priority : m_fused_actions) {
+#ifdef FUSER_VERBOSE
+         printf("starting recording %p\n", action_priority.first);
+#endif
+         action_priority.first->startRecording();
+      }
+      m_recording = true;
+   }
+
+   void stopRecording() {
+      for ( auto & action_priority : m_fused_actions) {
+         action_priority.first->stopRecording();
+      }
+      m_recording = false;
+   }
+
+   ///////////////////////////////////////////////////////////////////////////
+   /// @brief set preserveOrder mode
+   ///////////////////////////////////////////////////////////////////////////
+   void preserveOrder(bool preserveOrder) {
+      for (auto & action_priority : m_fused_actions) {
+         action_priority.first->preserveOrder(preserveOrder);
+      }
+      m_preserve_action_order = preserveOrder;
+   }
+
+
+   ///////////////////////////////////////////////////////////////////////////
+   /// @brief set scan mode
+   ///////////////////////////////////////////////////////////////////////////
+   virtual void setScan(bool scan) {
+      for (auto & action_priority : m_fused_actions) {
+         action_priority.first->setScan(scan);
+      }
+      m_is_scan = scan;
+   }
+
+   ///////////////////////////////////////////////////////////////////////////
+   /// @brief set counts_to_offsets_scan mode
+   ///////////////////////////////////////////////////////////////////////////
+   virtual void setCountsToOffsetsScan(bool scan) {
+      for (auto & action_priority : m_fused_actions) {
+         action_priority.first->setCountsToOffsetsScan(scan);
+      }
+      m_is_counts_to_offsets_scan = scan;
+   }
+
+
+   void flushActions(bool async) {
+#ifdef FUSER_VERBOSE
+      printf("Observer flushActions()\n");
+#endif
+      for (auto & priority_action : m_fused_action_order) {
+         FusedActions * const & actions = priority_action.second;
+#ifdef FUSER_VERBOSE
+         printf("Observer::flushActions::action count %i\n",actions->actionCount());
+#endif
+         if (actions -> actionCount() > 0) {
+#ifdef FUSER_VERBOSE
+            printf("Observer::flushActions priority:%g ptr:%p\n",priority_action.first, actions);
+#endif
+            actions->flushActions(async);
+         }
+      }
+      for (auto & array: m_to_be_freed) {
+         array.free();
+      }
+      m_to_be_freed.clear();
+   }
+
+   void registerFusedActions(FusedActions * actions, double priority) {
```

Comment:
Do we actually want all these methods to be inline?

Reply:
done

Reply:
I was really asking if having them inline is what we actually want. I wonder if having more loop fuser stuff implemented in the source file might reduce the compile times in places that include the loop fuser header.

Reply:
that's a good question. Can we make a ticket and pursue this as a follow up branch?

Reply:
Sure.